### ISSUE TITLE
the music don't go to the next song automatically

### ISSUE THREAD

**Issue by apierre55000** [jeffvli/feishin#550]
i listen songs from similarly songs generated and from coming's songs, but i'm obliged to double click on the next song because when the song is finished to play, it stops and the next song does'nt play automatically

i use the mpv conf player and the last feishin version 0.6.1 on windows 11 (the mpv version is 0.37)

**Comment by Fluttrr** [comment]
I'm having the same issue, but it happens everywhere. In playlist, albums, it does not automatically play the next song. When it's in this state where it finished a song and doesn't skip to the next one, I can't seek backwards in the current track either. The bar gets stuck at the selected time and the song doesn't play. This only happens with the web player though, using mpv works fine for me.

Feishin v0.6.1, Navidrome 0.50.2 (823bef54), Windows 11

**Comment by kgarner7** [comment]
> i listen songs from similarly songs generated and from coming's songs, but i'm obliged to double click on the next song because when the song is finished to play, it stops and the next song does'nt play automatically
> 
> i use the mpv conf player and the last feishin version 0.6.1 on windows 11 (the mpv version is 0.37)

When you double click a song in the `Related` tab, it is either enqueued (now/later) or played immediately in the `Up Next` tab. Is this the behavior you are seeing? Or is what you are seeing it will enqueue (or play, or whatever) the tracks but then it stops?

**Comment by ChekeredList71** [comment]
I can reproduce this bug and I have a solution too:

Feishin jumps to the next song, when:
- I click on the artist's top 1 song, it ends then the top 2 starts playing*

Feishin doesn't jump to the next song, when:
- the second top song is over (started playback like in the example marked with *)
- any album is played (from start or any other point)
- music is started with the artist playbutton

Clearing browser cache doesn't help either.

---

## Solution: 

On Debian 12 switching to mpv works. Any of the issues only appear with web audio player. 
(mpv 0.35.1, Feishin 0.6.1 AppImage, FFmpeg 5.1.4-0+deb12u1, Navidrome 0.51.1)

**Comment by kgarner7** [comment]
Is this still seen with web player? I'm not able to reproduce this behavior

**Comment by Fluttrr** [comment]
> Is this still seen with web player? I'm not able to reproduce this behavior

Yup, I switched back to the web player, played the top song from a smart playlist, skipped to the end and it failed to skip to the next song. Sometimes it does work though, and I really can't figure out why that would be at all. I'm not too bothered by this because using mpv works just fine but it does definitely still happen for me.

**Comment by kgarner7** [comment]
Ok...I'm not able to reproduce that at at all. Can you please try it on https://feishin.vercel.app (and have inspect element to show any errors)?

**Comment by Fluttrr** [comment]
This doesn't happen on the vercel app, however I've had this happen on both linux mint and a fresh installation on windows 11 just now. I also checked the console in the actual program (in windows) but there are no errors or warnings. I can't see any activity in the network tab when it's stuck at the end either.

**Comment by sprnza** [comment]
I does happen on the vercel app now to me.

**Comment by sprnza** [comment]
I believe it was working fine before the latest upgrade to 0.10.0

**Comment by jeffvli** [comment]
@sprnza I'm not seeing any errors on my end on the current version. Can you check the devtools console (F12) to see if you're getting any errors?

**Comment by sprnza** [comment]
@jeffvli   It appears to be an issue with qutebrowser  (qt6-webengine more specifically) as the issue is not reproducing in Chromium browser. There nothing appears on the devtools console except this which is already there when I open the Console.
![18-10-2024-17:31:44_grim](https://github.com/user-attachments/assets/28c55647-5d9d-41da-909d-b1d407870852)

**Comment by sprnza** [comment]
But Navidrome UI works fine though.

**Comment by vianraaa** [comment]
still happens on the 0.12.1 windows desktop client, no errors reported by devtools 
![image](https://github.com/user-attachments/assets/f9c70d6f-ab24-46b7-9380-c42debbb247f)

**Comment by sprnza** [comment]
The issue is still there with v0.20.0

**Comment by zeroquinc** [comment]
Yep, same problem here. When it does get stuck its always 3 seconds before the next song would play somehow. When I click next on the bar it continues.

**Comment by jeffvli** [comment]
For anyone who has this issue, I need more details, because I have not been able to reproduce this.

1. What environment are you using Feishin on? (Web / Desktop / Operating System)
2. What audio player are you using? (Web / MPV)
3. If using web:
    -  Is `use web audio` enabled?
    - Are you using `normal` or `crossfade`?
4. Are you able to reproduce this issue consistently? When you notice it happen with a certain song, try to reproduce it with the same song.

**Comment by zeroquinc** [comment]
> For anyone who has this issue, I need more details, because I have not been able to reproduce this.
> 
>     1. What environment are you using Feishin on? (Web / Desktop / Operating System)
> 
>     2. What audio player are you using? (Web / MPV)
> 
>     3. If using web:
>        
>        * Is `use web audio` enabled?
>        * Are you using `normal` or `crossfade`?
> 
>     4. Are you able to reproduce this issue consistently? When you notice it happen with a certain song, try to reproduce it with the same song.

1. Feishin, latest version, Windows 11 24H2
2. Web
3. Web audio is checked. I tried both normal and crossfade, happens on both. Sometimes it will play the next song but the playing bar is stuck on 4:09 / 4:12. Navidrome web also doesnt scrobble anymore. When I press next it actually plays the track thats already playing from the start again.

EDIT: I just noticed if I enable crossfade, it plays the next song but the bar will get stuck on "Street Runner" at 4:09. If I disable crossfade the next song never plays and it just gets stuck at 4:09 too.

4. Yes, I have one song I noticed that it always happens on. "Street Runner" by "Rod Wave". I actually just turned debugging on because I can reproduce it every time, and I see this happen in the Navidrome logs:

```
time="2025-09-24T22:27:58Z" level=warning msg="Request was interrupted" endpoint=/rest/stream.view error="context canceled" requestId=195a4069e040/NmBWOnBLPd-000033
```

I've also noticed this part never shows in the log when it happens:

```
time="2025-09-24T22:33:24Z" level=info msg="Now Playing" artist="Rod Wave" player="Feishin [feishin/Windows]" position=0 requestId=195a4069e040/NmBWOnBLPd-000095 title="Street Runner" user=desiler
time="2025-09-24T22:33:24Z" level=debug msg="Sending NowPlaying update" artist="Rod Wave" position=0 requestId=195a4069e040/NmBWOnBLPd-000095 scrobbler=lastfm track="Street Runner"
time="2025-09-24T22:33:24Z" level=debug msg="API: Successful response" endpoint=/rest/scrobble.view requestId=195a4069e040/NmBWOnBLPd-000095 status=OK
```

When it will get stuck it just goes straight to 

```
time="2025-09-24T22:28:52Z" level=debug msg="API: New request /rest/stream.view" client=Feishin requestId=195a4069e040/NmBWOnBLPd-000037 username=desiler version=1.13.0
time="2025-09-24T22:28:52Z" level=debug msg="Found matching player" client=Feishin id=WIhULRhhk0QMFsiLSoCLWn requestId=195a4069e040/NmBWOnBLPd-000037 type=feishin/Windows username=desiler
time="2025-09-24T22:28:53Z" level=debug msg="Streaming RAW file" id=wq9adY04Tvf1qctWyZVzxR originalBitrate=1004 originalFormat=flac path="/music/Michael Jackson/XSCAPE/17 - Love Never Felt So Good.flac" requestBitrate=0 requestFormat= requestId=195a4069e040/NmBWOnBLPd-000037 requestOffset=0 selectedBitrate=0 selectedFormat=raw
time="2025-09-24T22:28:53Z" level=info msg="Streaming file" artist="Michael Jackson & Justin Timberlake" bitRate=0 cached=false format=raw originalBitRate=1004 originalFormat=flac requestId=195a4069e040/NmBWOnBLPd-000037 title="Love Never Felt So Good" transcoding=false user=desiler
```

It seems like navidrome never receives a `New request /rest/scrobble.view` when it happens.

Also, when crossfade is enabled and the next song is playing when the bar is stuck, Navidrome WebUI doesn't report a scrobble. My LastFM also doesn't scrobble.

**Comment by jeffvli** [comment]
> Web audio is checked. I tried both normal and crossfade, happens on both. Sometimes it will play the next song but the playing bar is stuck on 4:09 / 4:12. Navidrome web also doesnt scrobble anymore. When I press next it actually plays the track thats already playing from the start again.

From your description, it seems like the same issue that https://github.com/jeffvli/feishin/pull/1095 was attempting to solve, which is a mismatch between the metadata duration and the actual duration of the file. For example, the file's metadata will return `4:12` as the duration, but is only `4:09` in reality. This causes an issue with the Web player implementation, since it uses 2 separate audio elements to attempt to seamlessly transition between tracks. If the track isn't able to reach the intended length `4:12`, it will become "stuck" and not transition to the next track.

Looking at the changes in the PR again, I realized that it only handles cases where the player errors out in some form. 

Could you check what Navidrome returns as the `duration` for that track? You can use the `Get Info` context menu item to check within Feishin.

**Comment by zeroquinc** [comment]
> From your description, it seems like the same issue that [#1095](https://github.com/jeffvli/feishin/pull/1095) was attempting to solve, which is a mismatch between the metadata duration and the actual duration of the file. For example, the file's metadata will return `4:12` as the duration, but is only `4:09` in reality. 

Yeah you could be right, because the song has a 12 second silence pause at the end.

Get Info returns: 
Duration | 4 min 12 sec

Not sure if it helps, but other media players also detect it as 4:12.

I just checked out the Spotify version, it has the same 12 seconds of silence at the end.

**Comment by jeffvli** [comment]
If you play the song (using your own local file) in another media player, can you check if you can actually scrub past 4:09?

**Comment by zeroquinc** [comment]
> If you play the song (using your own local file) in another media player, can you check if you can actually scrub past 4:09?

I checked it in foobar2000 and Windows Media Player and in both I can scrub past 4:09.

**Comment by jeffvli** [comment]
Thanks for the info. 

Seems like the solution here is to just trust the server duration as opposed to the one retrieved from the player itself.

I'm pushing a PR that should hopefully resolve this for good.

**Comment by jeffvli** [comment]
@zeroquinc Can you download/install the PR build from the `windows-binaries` on the build page and test to see if its resolved? https://github.com/jeffvli/feishin/actions/runs/17992499613?pr=1150

**Comment by zeroquinc** [comment]
> [@zeroquinc](https://github.com/zeroquinc) Can you download/install the PR build from the `windows-binaries` on the build page and test to see if its resolved? https://github.com/jeffvli/feishin/actions/runs/17992499613?pr=1150

Sadly it's not resolved, it's the exact same behavior as before installing this build.

**Comment by jeffvli** [comment]
Yeah, after I thought about it again, my changes would not have resolved the issue since it's getting stuck *before* the ending.

Could you send me the file so I can test it locally?

**Comment by zeroquinc** [comment]
> Yeah, after I thought about it again, my changes would not have resolved the issue since it's getting stuck _before_ the ending.
> 
> Could you send me the file so I can test it locally?

Ofcourse! https://filebin.net/36ussdeo9f02bqj0

Thanks for looking into it

**Comment by jeffvli** [comment]
Using your file I was able to improve the web player error handler which should gracefully transition the song when it runs into that specific error.

```
MediaError {code: 4, message: 'DEMUXER_ERROR_COULD_NOT_PARSE: FFmpegDemuxer: PTS is not defined'}
```

@kgarner7 Was also able to examine your file and determined that it's partially corrupted, which can probably be fixed by just re-encoding it.

Try out the new build from: https://github.com/jeffvli/feishin/actions/runs/17993283211?pr=1150

**Comment by zeroquinc** [comment]
> Using your file I was able to improve the web player error handler which should gracefully transition the song when it runs into that specific error.
> 
> ```
> MediaError {code: 4, message: 'DEMUXER_ERROR_COULD_NOT_PARSE: FFmpegDemuxer: PTS is not defined'}
> ```
> 
> [@kgarner7](https://github.com/kgarner7) Was also able to examine your file and determined that it's partially corrupted, which can probably be fixed by just re-encoding it.
> 
> Try out the new build from: https://github.com/jeffvli/feishin/actions/runs/17993283211?pr=1150

Thank you, it skipped to the next song now! 

Hmm, first time I hear that a file is corrupted. Any other media player that was connected to Navidrome played it fine (DSub, Symfonium). Which part was corrupted? 

Anyway, really glad that it's been fixed and so fast as well! Kudos to you guys.

**Comment by kgarner7** [comment]
[ffmpeg-reencode.log](https://github.com/user-attachments/files/22525751/ffmpeg-reencode.log)
Your file is truncated somehow (or has bad frames). I suspect most players just silently ignored, but it's obvious when I use ffmpeg to reencode the file

**Comment by zeroquinc** [comment]
> [ffmpeg-reencode.log](https://github.com/user-attachments/files/22525751/ffmpeg-reencode.log) Your file is truncated somehow (or has bad frames). I suspect most players just silently ignored, but it's obvious when I use ffmpeg to reencode the file

Thanks, just did a check on all my files and it found like 6 albums that are partially corrupted. Including every song of the Rod Wave album.
I'll try to re-encode them, thanks alot!
